pkgs/by-name/db/dbdev/package.nix

```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  darwin,
+  fetchFromGitHub,
+  gitUpdater,
+  openssl,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "dbdev";
+  version = "0.1.6";
+  sourceRoot = "${src.name}/cli";
+
+  src = fetchFromGitHub {
+    owner = "supabase";
+    repo = "dbdev";
+    rev = "v${version}";
+    hash = "sha256-G9U+dhSJI03n0lbBEVOrzslj8TlmiRwrg8DQ7hGsqIo=";
+  };
+
+  cargoHash = "sha256-v7QHUCKJjfTiG8ZylBmJ0rWRaeU9imGJBQhcESzZyUM=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  # Needed to get openssl-sys to use pkg-config.
+  env.OPENSSL_NO_VENDOR = 1;
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
```

Comment:
```suggestion
  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
```

Reply:
And I think you want to avoid the top-level darwin attribute

Reply:
> And I think you want to avoid the top-level darwin attribute

What would you suggest instead / why's that?

Many packages seem to be using top-level darwin including well known ones like dpkg:
https://github.com/NixOS/nixpkgs/blob/6ff5642249be62131aed96fc50111a9eea834f91/pkgs/by-name/dp/dpkg/package.nix#L17

Reply:
`darwin.apple_sdk.frameworks` is a package set. We also don't use `pkgs` as input and instead specify the individual packages. Though now that I think about it again, it is not that important here, since usually nothing in darwin is overriden.